### PR TITLE
[PAR-3817] models for shipping protection extension attribute object

### PR DESCRIPTION
**Pull request by scott-taylor-extend**
This is part of a larger change where we switch the order, invoice, creditmemo, and quote shipping protection extension attribute from an array to an object. I also reduced duplicate code by creating two helper functions in the ShippingProtectionTotalRepository.

### CODE REVIEW

File: Model/Sales/Order/Creditmemo/Total/ShippingProtection.php [hunk 1]
```diff
@@ -6,17 +6,22 @@
...
     private ShippingProtectionTotalRepositoryInterface $shippingProtectionTotalRepository;
+    private ShippingProtectionFactory $shippingProtectionFactory;
 
     public function __construct(
-        ShippingProtectionTotalRepositoryInterface $shippingProtectionTotalRepository
+        ShippingProtectionTotalRepositoryInterface $shippingProtectionTotalRepository,
+        ShippingProtectionFactory $shippingProtectionFactory
     ){
         $this->shippingProtectionTotalRepository = $shippingProtectionTotalRepository;
+        $this->shippingProtectionFactory = $shippingProtectionFactory;
```
Comment (jay-faust-extend) [inline Model/Sales/Order/Creditmemo/Total/ShippingProtection.php:24]: What is the factory doing here? Just needed for the base class?

**Review by jay-faust-extend — APPROVED** [review]

File: Model/ShippingProtection.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    /**
+     * Get SP Quote ID
+     *
+     * @return string
+     */
+    public function getSpQuoteId(): string
+    {
+        return $this->getData(self::SP_QUOTE_ID);
+    }
+}
```
Comment (jomarsantos) [inline Model/ShippingProtection.php]: nit: EOF newline?

**Review by jomarsantos — APPROVED** [review]

File: Model/ShippingProtection.php (hunk 2 above)
Comment (scott-taylor-extend) [reply Model/ShippingProtection.php]: fixed

File: Model/Sales/Order/Creditmemo/Total/ShippingProtection.php (hunk 1 above)
Comment (scott-taylor-extend) [reply Model/Sales/Order/Creditmemo/Total/ShippingProtection.php:24]: It's used on line 89

**Pull request merged**